packages/react-storage/src/components/StorageBrowser/createStorageBrowser/__tests__/createStorageBrowser.test-d.tsx

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable react/jsx-no-useless-fragment */
+import { expectAssignable, expectNotAssignable } from 'jest-tsd';
+import createStorageBrowser from '../createStorageBrowser';
+import { defaultActionConfigs } from '../../actions';
+import React from 'react';
+
+const { StorageBrowser, useAction, useView } = createStorageBrowser({
+  actions: {
+    default: {
+      ...defaultActionConfigs,
+    },
+    custom: {
+      myAction: {
+        handler: () => ({
+          result: Promise.resolve({ status: 'COMPLETE' }),
+        }),
+        viewName: 'MyActionView',
+        actionListItem: {
+          icon: 'info',
+          label: 'my action',
+        },
+      },
+    },
+  },
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  config: {} as any,
+});
+
+type StorageBrowserProps = Parameters<typeof StorageBrowser>[0];
```

Comment:
Why not use the actual `StorageBrowserProps` interface?

Reply:
This type contains the actual generics inferred from the `createStorageBrowser`. So we are testing against the type inferrence of `createStorageBrowser`.

Reply:
In that case we should provide a name that does not match the existing interface name to decrease the potential for confusion, example:
```ts
type TestStorageBrowserProps = Parameters<typeof StorageBrowser>[0];
```